pkg/storage/db.go

```diff
@@ -230,16 +257,25 @@ func (db *DB) GeneratePartition(ctx context.Context) error {
 		return handleRollback(tx, err)
 	}
 
+	beginTime, err := time.Parse(time.RFC3339, begin)
+	if err != nil {
+		return handleRollback(tx, fmt.Errorf("Invalid partition range: %s, %v", latest, err))
+	}
 	latestTime, err := time.Parse(time.RFC3339, latest)
 	if err != nil {
 		return handleRollback(tx, fmt.Errorf("Invalid partition range: %s, %v", latest, err))
 	}
 
+	if beginTime.After(db.now()) {
+		// a table has already been created in preparation for the future
+		return nil
+	}
+
 	return generatePartitionForTime(ctx, tx, db.now(), latestTime)
 }
 
-// GeneratePartitionWithTimestamp generates the partition based on the given time
-func (db *DB) GeneratePartitionWithTimestamp(ctx context.Context, begin time.Time) error {
+// generatePartitionWithTimestamp generates the partition based on the given time
+func (db *DB) generatePartitionWithTimestamp(ctx context.Context, begin time.Time) error {
```

Comment:
The integration tests use this. I am not sure why the CI didn't fail. Is `make integration` passing for you locally?